src/lexing_parsing/parser.ipp

```diff
@@ -276,11 +281,26 @@ private:
         return ast::Instruction(parseReturnStatement());
       case lexer::TT_K_ASM:
         return ast::Instruction(parseInlineAsmStatement());
+
+      /* case lexer::TT_K_INT: return ast::Instruction(...); */
+      /* case lexer::TT_K_VOID: return ast::Instruction(...); */
+      /* ... */
+      #define X(token, str) \
+        case token: return ast::Instruction(parseDeclaration());
+        PURE_TYPES_TOKEN_LIST
+      #undef X
       default:
         USER_THROW("Unexpected token while parsing instruction [" << _currentToken.type << "]", _currentToken.position);
     }
   }
 
+  ast::Declaration parseDeclaration()
+  {
+    ast::Type type = parseType();
+    std::string_view name = match(TT_IDENT);
```

Comment:
no initialisation yet I see, that's fine but we'll have to add in the future
name = expr;;
name{expr,...};
name(expr,...);